app/views/base/topnav.scala

```diff
@@ -15,7 +15,7 @@ object topnav:
   private def canSeeClasMenu(using ctx: PageContext) =
     ctx.hasClas || ctx.me.exists(u => u.hasTitle || u.roles.contains("ROLE_COACH"))
 
-  def apply()(using ctx: PageContext) =
+  def apply(me: lila.user.Me)(using ctx: PageContext) =
```

Comment:
Fun fact, `PageContext` already contains me. So if you want, you can revert that change to layout.scala remove the apply parameter and do `ctx.me.map(me => a(href := routes.Relation.following(me.username))(trans.friends())),`

Reply:
Ah ok, I haven't think about it. I just saw a similar approach with dasher component where me is being passed to dasher (it is in the same file). That's why I did it like that!

Reply:
I only knew that one because others correct my code all the time!

Reply:
`def apply(me: lila.user.Me)(using ctx: PageContext)` would mean that only logged in users can have a top nav menu.